metron-analytics/metron-profiler/src/main/java/org/apache/metron/profiler/bolt/ProfileBuilderBolt.java

```diff
@@ -243,8 +244,18 @@ private void flush(Tuple tickTuple) {
     measurement.setValue(result);
     emit(measurement, tickTuple);
 
+    // Execute the update with the old state
+    Map<String, String> tickUpdate = profileConfig.getTickUpdate();
+    Map<String, Object> state = executor.getState();
+    if(tickUpdate != null) {
+      tickUpdate.forEach((var, expr) -> executor.assign(var, expr, executor.getState()));
```

Comment:
The 3rd argument to executor.assign is called 'transientState'.  It exists only to make additional state available to the expression;  additional state that is not already maintained by the executor itself.  The normal use for it is when you want to make the fields in a telemetry message available to the expression.

Setting transientState=executor.getState() does not make sense here.  It is exactly the same as passing nothing at all.  The state in executor.getState() is already exposed to the expression.

Hopefully, I am describing this well enough.  As another feeble attempt at explanation, the following two expression are equivalent.

```
tickUpdate.forEach((var, expr) -> executor.assign(var, expr, executor.getState()));
```
```
tickUpdate.forEach((var, expr) -> executor.assign(var, expr, null));
```